addons/hr_holidays/tests/test_automatic_leave_dates.py

```diff
@@ -165,8 +169,8 @@ def test_attendance_on_morning(self):
             # Ask for afternoon
             leave_form.request_date_from_period = 'pm'
 
-            self.assertEqual(leave_form.number_of_days_display, 1)
-            self.assertEqual(leave_form.number_of_hours_text, '8 Hours')
+            self.assertEqual(leave_form.number_of_days_display, 0)
+            self.assertEqual(leave_form.number_of_hours_text, '0 Hours')
```

Comment:
It's weird that this shows up on this PR because I think it's this PR that made that change: https://github.com/odoo/odoo/pull/71135
It's basically to ensure that if you take leaves outside of your working schedule (which shouldn't happen), you get zero duration. This needed to be fixed because of the hacky way half-day durations were computed.